api/src/main/java/tennisclub/facade/CourtFacade.java

```diff
@@ -0,0 +1,17 @@
+package tennisclub.facade;
+
+import tennisclub.dto.CourtDto;
+import tennisclub.enums.CourtType;
+import java.util.List;
+
+/**
+ * @author Pavel Tobiáš
+ */
+public interface CourtFacade {
+    void create(CourtDto court);
```

Comment:
I am not sure whether the argument should be `CourtDto`. When I looked at the reference implementation, they had another DTO _(in your case it would be a `CourtCreateDTO` - it is almost identical to the CourtDTO, but it lacks the `id` attribute because when you create a new instance, you create it without its `id`)_. Personally, I am not very fond of the DTO pattern so personally I would approve but I am not sure whether the seminar tutor would like it 😁.

Reply:
@Fuyunori , let me add it then. Do you think would be a problem if `CourtDto extends CourtCreateDto` to reduce the amount of boilerplate code?

Reply:
@paveltobias I was actually researching today whether it is good to have a DTO object inherit from another. I haven't seen any convincing answers, however based on what I have seen, most of them discourage from doing so _(but again, personally I would also prefer inheritance in this case)_. So I myself don't really know answer to this question - we can ask about this on the seminar 😁

Reply:
@Fuyunori, oops, already made the change using inheritance. :laughing: Never mind, let me correct it. :smile:

Reply:
@paveltobias you can leave it as it is 😁😀

Reply:
@Fuyunori, lol, we are somewhat out of sync; already did the fix. :laughing: Anyhow, the PR should be mergeable now.

Reply:
@Fuyunori, ...but let us ask the tutor anyway. Do you still have links to the sources discouraging from using inheritance between DTOs?

Reply:
@paveltobias Thanks and good job 😊👏 Also I think the DTO might require having `equals` and `hashCode` _(or at least the reference implementation has it, but I'm not sure what is the correct approach there)_. 😁

Reply:
@paveltobias this is the probably the most convincing one: https://stackoverflow.com/questions/37870918/should-dtos-use-inheritance-or-composition - especially this sentence:

> The "one DTO class" approach is almost certainly bad. It smells like a God Class. Many pundits excoriate DTOs altogether. You could inherit from some base class, but for value objects it's not really sensible. Same for composition. It makes your code more complicated.

Because it's a value class, it just simply doesn't make much sense to use inheritance 😐

Reply:
@paveltobias there is another link: https://softwareengineering.stackexchange.com/questions/399227/is-it-anti-pattern-to-have-inheritence-in-a-dto

> It sounds kind of funny to me when I read it out loud. Inheritance in DTOs becomes problematic, because you introduce coupling between classes where each class usually represents a concrete idea or use case. DTOs sharing a common parent class implies the class hierarchy will evolve at the same time, and for the same reasons. This makes these classes harder to change without having a ripple effect cascading to classes where the new fields might not be applicable.

Reply:
@paveltobias semantically it doesn't make much sense _(CourtDTO IS NOT a CourtCreateDTO)_ - doesn't comply with the is-A relationship _(inheritance)_.

Reply:
@Fuyunori, `equals()` and `hashCode()` generated for both DTOs. :heavy_check_mark:

Reply:
@Fuyunori Thank you for the links and arguments! I will read through them and think about it. :smiley: